tests/update_workflow_test.go

```diff
@@ -1816,8 +1817,10 @@ func (s *integrationSuite) TestUpdateWorkflow_FailWorkflowTask() {
 			// Returning error will cause the poller to fail WT.
 			return nil, errors.New("malformed request")
 		case 4:
-			// 3rd attempt doesn't have any updates attached to it because UpdateWorkflowExecution call has timed out.
-			s.Empty(task.Messages)
+			// 3rd attempt UpdateWorkflowExecution call has timed out but the
+			// update is still running
+			updRequestMsg := task.Messages[0]
+			s.EqualValues(9, updRequestMsg.GetEventId())
```

Comment:
When the update was cleared/deleted at the end of the RPC handler function (with a defer), this was correct. However as part of supporting async updates, we will need to let Updates exist in the Registry longer than the RPC handler function is on the stack. Thus this modification to the test to expect that the update has _not_ gone away at this point.